api/server/router/container/backend.go

```diff
@@ -15,7 +13,7 @@ import (
 // execBackend includes functions to implement to provide exec functionality.
 type execBackend interface {
 	ContainerExecCreate(config *types.ExecConfig) (string, error)
-	ContainerExecInspect(id string) (*exec.Config, error)
+	ContainerExecInspect(id string) (interface{}, error)
```

Comment:
As you put it, JSONification of objects is indeed sufficiently magic. [`encoding/json`](http://golang.org/pkg/encoding/json) can serialize any Go value (except for functions and cyclic structures).

I opted for this approach, because the other would have meant copy-pasting both [`daemon/exec.Config`](https://godoc.org/github.com/docker/docker/daemon/exec#Config) and [`daemon/execdriver.ProcessConfig`](https://godoc.org/github.com/docker/docker/daemon/execdriver#ProcessConfig). The problem here is `execdriver.ProcessConfig`, which is inherently platform- and execdriver-specific, and therefore different depending on the build tags used.

I though that in this case, using `interface{}` for an essentially opaque value was the lesser of two evil, compared to duplicating highly platform-dependent code.